File: branches/soon/src/network/Protocol.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package network;

/**
 *
 * @author master
 */
public class Protocol {    
    // Constant define a timeouts
    public static final int TIMEOUT_CONNECTION = 20;
    public static final int TIMEOUT_ACCEPT = 100;
    
    // Symbol delimiters of protocol
    public final static String COM_DELIMITER = "`";
    public final static String ARGS_DELIMITER = "~";
    
    // Answers of protocol
    public final static String OK = "ok";
    public final static String NO = "no";
    
    // Commands of protocol
    public final static String ERROR = "err";
    public final static String IS_END = "end";
    public final static String CLOSE = "cl";
    public final static String PIN_INIT = "p_in";
    public final static String PIN_EXEC = "p_ex";
}

File: soon/PinServer/protocol.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "protocol.hpp"

void reverseBytes(byte* arr, int size) {
    byte tmp;
    int half_size = size / 2;
    for(int i = 0; i < half_size; i++) {
        tmp = arr[i];
        arr[i] = arr[size - i - 1];
        arr[size - i - 1] = tmp;
    }
}